core/src/layers/prometheus.rs

```diff
@@ -645,11 +717,16 @@ impl<R: oio::Read> oio::Read for PrometheusMetricWrapper<R> {
     }
 
     fn poll_next(&mut self, cx: &mut Context<'_>) -> Poll<Option<Result<Bytes>>> {
+        let labels = self.stats.generate_metric_label(
```

Comment:
Maybe we can store labels in `PrometheusMetricWrapper` so that we don't need to build them every time calling `poll_read`.

Reply:
I'm not sure if there is some best practice in Rust to store the `Vec<&str>` in struct

If we store the Vec<String> instead, there will be two conversion when use it(convert vec<&str> to Vec<String> when input, and convert back when output), I not sure it would be faster than build them every time

Reply:
This is not a blocker. We can improve this part later.

Reply:
Nice